### PR TITLE
DOC-595: Improve create index

### PR DESCRIPTION
**Pull request by diana-macrometa**
- Added content from Query Language Enhancements into the existing doc.
- Added example explanations.
- Added crosslinks.
- Other minor edits.

### CODE REVIEW

File: docs/cep/table/table-indexes.md [hunk 1]
```diff
@@ -3,34 +3,172 @@ sidebar_position: 30
...
+| fields_to_index | Comma separated list of 1:* index fields without types (i.e id, name). |
+
+### CREATE SOURCE with an Index
+
+You can incorporate it into a source definition for a `database` (Macrometa collection) source. For more information about source definitions, refer to [Sources](../source/index.md).
 
 ```sql
--- Creates a persistent index named `SamplePersistentIndex` on `SampleGDNTable` with following properties {unique=true, sparse=true, deduplicate=true}.
-CREATE UNIQUE INDEX SamplePersistentIndex ON TABLE SampleGDNTable WITH(type="persistent", sparse="true", deduplicate="true") (sensorId);
+CREATE SOURCE <source_name> WITH (type="database", collection="<collection>", PrimaryKey="<primary-key-field>", Index="<index-name-1>", Index="<index-name-2>"...) (<attribute_name> <attribute_type>, <attribute_name> <attribute_type>...);
```
Comment (diana-macrometa) [inline docs/cep/table/table-indexes.md]: @StoyanTc, what kind of index is created if this method is used? Can you specify an index type with this method?

File: docs/cep/table/table-indexes.md [hunk 2]
```diff
@@ -3,34 +3,172 @@ sidebar_position: 30
...
+```
 
--- Creates a hash index named `SampleHashIndex` on `SampleGDNTable` with properties {unique=true, sparse=true, deduplicate=true}.
-CREATE UNIQUE INDEX SampleHashIndex ON TABLE SampleGDNTable WITH(type="hash", sparse="true", deduplicate="true") (sensorId);
+| Parameter           | Description                 |
```
Comment (diana-macrometa) [inline docs/cep/table/table-indexes.md]: @StoyanTc, please check this table carefully, it is new.

File: docs/cep/table/table-indexes.md [hunk 3]
```diff
@@ -3,34 +3,172 @@ sidebar_position: 30
...
+## Example 1
+
+```sql
+CREATE TABLE SampleGDNTable (sensorId string, temperature double);
+CREATE UNIQUE INDEX SamplePersistentIndex ON TABLE SampleGDNTable WITH(type="persistent", sparse="true") (sensorId);
+```
+
+This example creates a table named `SampleGDNTable` with two fields: `sensorId` of type `string` and `temperature` of type `double`. The second statement creates a unique, persistent index named `SamplePersistentIndex` on the `SampleGDNTable` table with `sensorId` as the field to index.
+
+## Example 2
```
Comment (diana-macrometa) [inline docs/cep/table/table-indexes.md]: @StoyanTc - Is the explanation for this example correct? Also, what sort of index is created here?

File: docs/cep/table/_category_.json [hunk 4]
```diff
@@ -1,4 +1,4 @@
 {
-  "label": "Tables",
+  "label": "Tables (Collection)",
```
Comment (StoyanTc) [inline docs/cep/table/_category_.json]: `Collections` plural?

File: docs/cep/table/table-indexes.md (hunk 1 above)
Comment (StoyanTc) [reply docs/cep/table/table-indexes.md]: @diana-macrometa - This is related for a RDBMS table. The `c8db` collection does not have explicit PK. I would suggest to remove the section `CREATE SOURCE with an Index`.

File: docs/cep/table/table-indexes.md (hunk 3 above)
Comment (StoyanTc) [reply docs/cep/table/table-indexes.md]: @diana-macrometa - Example 2 is invalid.

**Review by StoyanTc — CHANGES_REQUESTED** [review]

**Review by StoyanTc — APPROVED** [review]

**Pull request merged**